packages/nimble-components/src/icon-base/tests/icons.mdx

```diff
@@ -1,26 +1,28 @@
-import {
-    Meta,
-    Controls,
-    DocsStory,
-    Title,
-    Description
-} from '@storybook/blocks';
+import { Meta, Controls, DocsStory, Title } from '@storybook/blocks';
 import * as iconsStories from './icons.stories';
+import { iconAddTag } from '../../icons/add';
 
 <Meta of={iconsStories} />
 
 <Title of={iconsStories} />
-<Description of={iconsStories} />
+
+Nimble icons can be slotted into other components or used independently. Each icon is available as a custom element. For example, <code>&lt;{iconAddTag}&gt;&lt;/{iconAddTag}&gt;</code>.
```

Comment:
I'm using the existing pattern where we use the `iconAddTag` value, instead of explicitly writing the tag name, but I see 2 issues:
1. The markup is gross and much less clear.
2. Storybook adds another styling class to the "backtick to `<code>` conversion", so explicitly using the `<code>` tag looks slightly different.

I'd prefer clearer markup vs. dynamism, but would be interested in a clear policy one way or the other.

Reply:
Can we make a helper mdx component for that? Pass the tagname to a `<TagName name={iconAddTag}>`?

Reply:
How about something like:

```jsx
/**
 * Renders a "TagName" component for Storybook documentation.
 */
export const TagName = ({ name, displayStyle }) => {
    switch (displayStyle) {
        case 'open-tag':
            return (<code>&lt;{name}&gt;</code>);
        case 'close-tag':
            return (<code>&lt;/{name}&gt;</code>);
        case 'open-close-tag':
            return (<code>&lt;{name}&gt;&lt;/{name}&gt;</code>);
        case 'self-closing-tag':
            return (<code>&lt;{name}/&gt;</code>);
        default:
            return (<code>{name}</code>);
    }
};
```

Which when called like:
```jsx
<p>
    <TagName name={iconAddTag} displayStyle={'open-tag'} /> <br />
    <TagName name={iconAddTag} displayStyle={'close-tag'} /> <br />
    <TagName name={iconAddTag} displayStyle={'self-closing-tag'} /> <br />
    <TagName name={iconAddTag} displayStyle={'open-close-tag'} /> <br />
    <TagName name={iconAddTag} />
</p>
```

Renders:
<img width="327" alt="Screenshot 2024-01-31 at 1 40 29 PM" src="https://github.com/ni/nimble/assets/1458528/6cfcf650-d9b5-4c19-bfb8-cae51957065f">

Reply:
looks good, maybe we can make it less verbose?

`mode` or `display` // mainly want to avoid the term "style"
open, close, self-close, open-close

Reply:
Maybe even separate attributes:
```
<TagName name={iconAddTag} />
<TagName name={iconAddTag} open />
<TagName name={iconAddTag} open-close />
<TagName name={iconAddTag} close />
<TagName name={iconAddTag} self-close />
```
Think that's possible in React Components, don't actually know

Reply:
Ah yea, they do support boolean props, see `#2` on https://www.freecodecamp.org/news/react-props-cheatsheet/

Reply:
maximum minimum :P 
```
<Tag name={iconAddTag} />
<Tag name={iconAddTag} open />
<Tag name={iconAddTag} open-close />
<Tag name={iconAddTag} close />
<Tag name={iconAddTag} self-close />
```

Reply:
Implemented and rolled it out to all usages - except for usage in multi-line code blocks, and where the strings were further concatenated with other things (CSS selectors)

Reply:
It'd be good to document the new block in the README for this folder. https://github.com/ni/nimble/blob/main/packages/nimble-components/.storybook/blocks/README.md?plain=1

Reply:
Done as –

```md
### Tag

-   Renders a tag string in code blocks. Optionally surrounding it with angle brackets to show as HTML elements.

    | Property  | Example                                 |
    | --------- | --------------------------------------- |
    | _default_ | `nimble-component`                      |
    | open      | `<nimble-component>`                    |
    | openClose | `<nimble-component></nimble-component>` |
    | close     | `</nimble-component>`                   |
    | selfClose | `<nimble-components/>`                  |
```